src/core/net/tcp6.cpp

```diff
@@ -95,40 +125,69 @@ const SockAddr &Tcp::Endpoint::GetPeerAddress(void) const
 
 Error Tcp::Endpoint::Bind(const SockAddr &aSockName)
 {
-    OT_UNUSED_VARIABLE(aSockName);
+    Error error;
+    VerifyOrExit(!static_cast<const Address *>(&aSockName.mAddress)->IsUnspecified(), error = kErrorInvalidArgs);
+    VerifyOrExit(GetInstance().Get<Tcp>().CanBind(aSockName), error = kErrorInvalidState);
 
-    return kErrorNotImplemented;
+    memcpy(&mTcb.laddr, &aSockName.mAddress, sizeof(mTcb.laddr));
+    mTcb.lport = HostSwap16(aSockName.mPort);
```

Comment:
I suggest adding 
```c++
using ot::Encoding::BigEndian::HostSwap16;
using ot::Encoding::BigEndian::HostSwap32;
```
at the top of this file to ensure we are using the big-endian here.

Without this I think we are indirectly relying on the `using` from `ip6_header.hpp`. I would say it may be better to be explicit about it at the top of the file.

Reply:
OK, I added those `using` statements.